### PR TITLE
Processing data in models

### PR DESCRIPTION
**Pull request by Feliverse**
### Create models
-  Create model classes for all entities as shown in the [ERD diagram](../images/blog_app_erd_v1_1.png).
-  Set up associations between models.
    - Remember that `author_id` column in `posts` table should be the foreign_key for the `users` table.

### Use models to insert data
- Familiarize yourself with ['rails c' command](https://guides.rubyonrails.org/command_line.html#bin-rails-console).
- Open rails console for your project.
- Create at least one user by running the following code:
    ```ruby
       first_user = User.create(name: 'Tom', photo: 'https://unsplash.com/photos/F_-0BxGuVvo', bio: 'Teacher from Mexico.')
       second_user = User.create(name: 'Lilly', photo: 'https://unsplash.com/photos/F_-0BxGuVvo', bio: 'Teacher from Poland.')
    ```
- Create at least 4 posts written by one of the users you created by running the following code:
    ```ruby
       first_post = Post.create(author: first_user, title: 'Hello', text: 'This is my first post')
    ```
- Create at least 6 posts comments for one of the posts you created by running the following code:
    ```ruby
       Comment.create(post: first_post, author: second_user, text: 'Hi Tom!' )
    ```
- Use other [CRUD methods](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/lessons/active_record_crud.md) to find, update and delete entities.

### Create custom methods
Your models should include some additional methods.
-  Users
    - A method that returns the 3 most recent posts for a given user.
-  Posts
    - A method that updates the posts counter for a user.
    - A method which returns the 5 most recent comments for a given post.
-  Comments
    - A method that updates the comments counter for a post.
-  Likes
    - A method that updates the likes counter for a post.
- Go to `rails c` and check if your methods are working.

### Need a big picture?

Remind me about [the big picture of this project](../sneak_peek.md).

### CODE REVIEW

File: app/models/post.rb [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+  has_many :likes
+
+  def update_counter
+    author.increment!(:post_counter)
+  end
+
+  def last_5_comments
+    comments.order(created_at: :desc).limit(5)
+  end
+end
```
Comment (alexander16108) [inline app/models/post.rb]: Optional - i suggest you use one of the rails active callbacks `after_save` to call the update counter method each time a new post has been saved  to the database 😄 and remember to make the methods private

File: app/models/like.rb [hunk 2]
```diff
@@ -0,0 +1,8 @@
+class Like < ApplicationRecord
+  belongs_to :user
```
Comment (alexander16108) [inline app/models/like.rb]: as seen in the ERD diagram the model should belong to the author as seen in your post.rb file with a class name of `user` 👍🏾

File: app/models/comment.rb [hunk 3]
```diff
@@ -0,0 +1,8 @@
+class Comment < ApplicationRecord
+  belongs_to :user
```
Comment (alexander16108) [inline app/models/comment.rb]: as seen in the ERD diagram the model should belong to the author as seen in your post.rb file with a class name of `user` 👍🏾

File: app/models/comment.rb [hunk 4]
```diff
@@ -0,0 +1,8 @@
...
+  belongs_to :post
+
+  def update_counter
+    post.increment!(:comment_counter)
+  end
+end
```
Comment (alexander16108) [inline app/models/comment.rb]: Optional - i suggest you use one of the rails active callbacks after_save to call the update counter method each time a new post has been saved to the database 😄 and remember to make the methods private

File: app/models/like.rb [hunk 5]
```diff
@@ -0,0 +1,8 @@
...
+  belongs_to :post
+
+  def update_counter
+    post.increment!(:like_counter)
+  end
```
Comment (alexander16108) [inline app/models/like.rb]: Optional - i suggest you use one of the rails active callbacks after_save to call the update counter method each time a new post has been saved to the database 😄 and remember to make the methods private

File: README.md [hunk 6]
```diff
@@ -21,4 +21,97 @@ Things you may want to cover:
 
 * Deployment instructions
```
Comment (alexander16108) [inline README.md]: please this feature are template required for your to know how to make a good readme file and should not be visible here please remove and update your readme accordingly

**Review by alexander16108 — CHANGES_REQUESTED** [review]
Hi @Feliverse , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



## Highlights

- good pr
- passing linters
- good readme file

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/models/post.rb (hunk 1 above)
Comment (Feliverse) [reply app/models/post.rb]: Nice suggestion, I will apply

File: app/models/comment.rb (hunk 4 above)
Comment (Feliverse) [reply app/models/comment.rb]: Nice suggestion, I will apply

File: app/models/like.rb (hunk 5 above)
Comment (Feliverse) [reply app/models/like.rb]: Nice suggestion, I will apply

File: README.md (hunk 6 above)
Comment (Feliverse) [reply README.md]: Thanks for pointing it out to me, I'll fix it now

File: app/models/comment.rb (hunk 3 above)
Comment (Feliverse) [reply app/models/comment.rb]: Thanks for pointing it out to me, I'll fix it now

File: app/models/like.rb (hunk 2 above)
Comment (Feliverse) [reply app/models/like.rb]: Thanks for pointing it out to me, I'll fix it now

**Review by KayLemba — APPROVED** [review]
[[# STATUS: APPROVED :green_circle: :clap: 

Hello @Feliverse          , 
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉  Keep up the good work 🥂 

HIGHLIGHTS:

- [x]  Your repo is professional  😎 
- [x] Descriptive P.R 💯 
- [x] linters are passing
- [x] Professional README 💯
- [x] Data in models setup correctly 💯 
 










 








Now you can merge this branch and move on :1st_place_medal: 
Happy Coding cheers 🥇 💯 



well-1

Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better.

[OPTIONAL] It would be nice to add your coding partners contact infomation to the Authors section because this is a pair programming exercise.

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._]]

**Pull request merged**